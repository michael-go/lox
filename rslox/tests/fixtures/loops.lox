var i = 0;
while (i < 4) {
    print "while looper";
    print i;
    i = i + 1;
}

for (; i > 0;) {
    print "naked for looper";
    print i;
    i = i - 1;
}

for (var j = 0; j <= 2; j = j + 1) {
    print "for looper";
    print j;
}