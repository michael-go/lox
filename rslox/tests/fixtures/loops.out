# exit code: 0
# stdout:
"while looper"
0
"while looper"
1
"while looper"
2
"while looper"
3
"naked for looper"
4
"naked for looper"
3
"naked for looper"
2
"naked for looper"
1
"for looper"
0
"for looper"
1
"for looper"
2

# stderr:

